### PR TITLE
Multi buf/v3

### PR DESCRIPTION
**Pull request by victorjulien**
https://redmine.openinfosecfoundation.org/issues/5784

suricata-verify-pr: 1183

replaces #8740

### CODE REVIEW

File: src/detect-urilen.c [hunk 1]
```diff
@@ -147,86 +147,96 @@ static void DetectUrilenFree(DetectEngineCtx *de_ctx, void *ptr)
...
-            cd->flags |= DETECT_CONTENT_DEPTH;
-            SCLogDebug("updated %u, content %u to have depth %u "
-                    "because of urilen.", s->id, cd->id, cd->depth);
+        // skip 65535 to avoid mismatch on uri > 64k
+        if (!found || high == UINT16_MAX)
+            return;
+
+        SCLogDebug("high %u", high);
+
+        sm = s->init_data->smlists[list];
```
Comment (victorjulien) [inline src/detect-urilen.c:197]: this is wrong

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on ASAN_TLPR1_cfg.
  

Pipeline 13272

**Comment by ct0br0** [comment]
Re-ran with the single rule update. Everything looks good. Pipe 13479 in dev

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 113274 | 145034 | 128.04% | 

Pipeline 13523

**Comment by victorjulien** [comment]
Merged in #8792

**Comment by victorjulien** [comment]
See https://redmine.openinfosecfoundation.org/issues/6032

**Pull request closed without merging**